user_guide/user_guide.md

```diff
@@ -250,9 +250,115 @@ In the editing mode you can also delete existing Functions.
+
+<br/>
+<kbd>
+<img src="../resources/images/create_api2.png" width="350">
+</kbd>
+<br/>
+
+**Placeholder ID** gets defined automatically and is needed when using the placeholder in other fields.
+
+**Placeholder Field** specifies the field of the Function which is used to replace the placeholder. You can select in a dropdown menu every field of the Function you defined in the "Belongs to Function" field.
+
+
+Alternatively to replacing the Placeholder by a Function field it can also be replaced by the result of another Function. *Check the "Replace Placeholder by value of the function this API belongs to." box to switch and click again to switch back.*
+
+<br/>
+<kbd>
+<img src="../resources/images/create_api3.png" width="350">
+</kbd>
+<br/>
+
+**Placeholder Function name** specifies the Function the Placeholder receives the result from.
+
+**Placeholder Function Fields** can be set if the Function needs some fields to return the value.
+
+Some API need the values in a String format or in a number format. *Uncheck the "Do not convert placeholder value to 'String' before insertion into a JSON template." checkbox to convert the Placeholder value to a string, check to leave the orginal value.*
+
+You can add **Headers** to the API request.
+*Click the "Add a new key-value pair", bin and "Delete all key-value pairs" buttons to add and remove header fields*
+
+<br/>
+<kbd>
+<img src="../resources/images/create_api4.png" width="350">
+</kbd>
+<br/>
+
+In the "Provide a template for required URL parameter" field you can add **Parameters** in the JSON format to the API request.
+
+In the "Provide a template for the outgoing requests" field you can add a **Body** in the JSON format to the API request.
+
+**Response Path to Result** specifies how the API result is retrieved from the response body. The path is defined by a series of JSON keys that are seperate with a `.`. Elements in lists can be accessed via their index in square brackets `[index]`. The sample response path `devices.lamps[0].power` takes the value under the "devices" key, then the list under the "lamps" key, then the 0. element of that list and then the value under "power" key.
+
+**Request Method** specifies which HTTP request method is used for the API request which can be selected from the dropdown menu.
+
+We show an example API definition, the Finnhub Finance API which translates a custom string to a company symbol with the Company Symbol Function.
+<br/>
+<kbd>
+<img src="../resources/images/create_api5.png" width="450">
+<img src="../resources/images/create_api6.png" width="450">
+</kbd>
+<br/>
+
+Let us put attention on how the Placeholder was used to call another Function with the input i.e. field of the original function and put the result of that Function in the API body template.
```

Comment:
**Genauer:**
Let us put attention on how the placeholder is evaluated by calling another function named "retrieve_company_symbol" and passing the value of the field "company_name" of the "stock_price" function to which this API belongs. The result of that function call, i.e. the company_symbol, is finally the value of this placeholder and thus inserted into the template given for the URL parameters.